src/pay.rs

```diff
@@ -170,17 +170,20 @@ impl Runtime {
             .as_ref()
             .or(iface.default_operation.as_ref())
             .ok_or(CompositionError::NoOperation)?;
+
         let assignment_name = invoice
             .assignment
             .as_ref()
             .or_else(|| {
-                iface
-                    .transitions
-                    .get(operation)
-                    .and_then(|t| t.default_assignment.as_ref())
+                iface.transitions.get(operation).and_then(|t| {
+                    t.default_assignment
+                        .clone()
```

Comment:
why do we need clone here?

Reply:
Without clone, i receive the compile error below:

```cannot move out of `t.default_assignment` which is behind a shared reference```

Reply:
Do `as_ref` instead of clone

Reply:
Done